### DISCUSSION TITLE
Evaluation multilabel text classification - weighted, samples

### DISCUSSION THREAD

**Discussion by hp52** [explosion/spaCy#8313] [Help: Coding & Implementations] [answered]
Hi,
I'd like to evaluate multi label text classification. I find implementations in Spacy for micro and macro F-Score/Precision/Recall. What about "weighted" and "samples"? Are those metrics implemented in Spacy?
Thanks!

**Comment by svlandeg** [comment, marked as answer]
You can find some related discussion threads here: 
https://github.com/explosion/spaCy/discussions/8125
https://github.com/explosion/spaCy/discussions/8258
https://github.com/explosion/spaCy/discussions/8226

and the currently implemented scoring functions are documented here: https://spacy.io/api/scorer

In short, there's currently no built-in mechanism for weighted scores, but you can overwrite `TextCategorizer.score`. 

It's on our roadmap to make this more extensible/configurable: https://github.com/explosion/spaCy/pull/8256
